ash/src/entry.rs

```diff
@@ -46,6 +44,22 @@ pub enum LoadingError {
     LibraryLoadError(String),
 }
 
+impl fmt::Display for LoadingError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "LoadingError::{:?}", self)
```

Comment:
The `Display` impl should not be equal to the `Debug` impl; people can just use `Debug` for that. Instead, it should provide a written description of the error. In this case, probably just print the wrapped string?

Reply:
As stated below, the debug impl does not render the constant name for some things (specifically `Result`). The name of the constant is more valuable IMO, as you can plug that into google. I do like having the comments from the xml added to the rustdoc though.

I would prefer to show the constant name for display and the type and value for debug.

e.g.

Display: `VK_SUCCESS`
Debug: `Result(0)`

Reply:
This comment refers to `LoadingError`, not `vk::Result`. There are no constants here.

Reply:
I followed the convention that was established by the crate's author:

```rust
impl fmt::Display for InstanceError {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "InstanceError::{:?}", self)
    }
}
```

Reply:
In this case, I believe that's a poor example to follow. `Display` impls for errors in rust should always produce a written description of the error; this convention is well established. In this case, unlike `InstanceError`, it should be completely trivial to do so since there's just the one wrapped string anyway.

Reply:
It's not one wrapped string. The other error type contains a `Vec`.

```rust
pub enum InstanceError {
    LoadError(Vec<&'static str>),
    VkError(vk::Result),
}
```  

I've changed them both to print the inner value though, based on this feedback.

Reply:
Personally, I think the author's way is better overall. I'm happy to rip off the last commit to revert this:

https://github.com/MaikKlein/ash/pull/194/commits/8cb149e94b19aff9e5f156461b60412740586d12

Reply:
> It's not one wrapped string

This thread is discussing your change to `LoadingError`, which is always exactly one wrapped string.

Reply:
They should be consistent.

Reply:
@MaikKlein Do you have a preference here? Does this need further changes?

Reply:
I think we should just print the string. I'll add my comments on Debug/Display in the other discussion.

Reply:
> I've changed them both to print the inner value though, based on this feedback.

```rust
impl fmt::Display for LoadingError {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        match self {
            LoadingError::LibraryLoadError(e) => write!(f, "{}", e),
        }
    }
}```